File: src/main/java/com/extendedae_plus/common/init/ModCreativeTabs.java
package com.extendedae_plus.common.init;

import com.extendedae_plus.ExtendedAEPlus;
import com.extendedae_plus.common.registry.dataComponent.DataTickingCard;
import com.extendedae_plus.util.UtilTextComponent;
import net.minecraft.core.registries.Registries;
import net.minecraft.network.chat.Component;
import net.minecraft.world.item.CreativeModeTab;
import net.neoforged.neoforge.registries.DeferredHolder;
import net.neoforged.neoforge.registries.DeferredRegister;

import java.util.stream.Stream;

public final class ModCreativeTabs {
    @InitObject
    public static final DeferredRegister<CreativeModeTab> TABS =
            DeferredRegister.create(Registries.CREATIVE_MODE_TAB, ExtendedAEPlus.MODID);

    public static final DeferredHolder<CreativeModeTab, CreativeModeTab> MAIN = TABS.register("main",
            () -> CreativeModeTab.builder()
                    .icon(ModItems.WIRELESS_TRANSCEIVER::toStack)
                    .displayItems((params, output) -> {
                        ModItems.ITEMS.forEach(output::accept);
                        Stream.of(
                                new DataTickingCard(2, 16),
                                new DataTickingCard(4, 192),
                                new DataTickingCard(8, 512),
                                new DataTickingCard(16, 1024)
                        ).map(DataTickingCard::toStack).forEach(output::accept);
                    }).withTabFactory(builder -> new CreativeModeTab(builder) {
                        @Override
                        public Component getDisplayName() {
                            return UtilTextComponent.modNameColorful;
                        }
                    }).build());
}

File: src/main/kotlin/com/extendedae_plus/common/init/ModCreativeTabs.kt
package com.extendedae_plus.common.init

import com.extendedae_plus.ExtendedAEPlus
import com.extendedae_plus.common.registry.dataComponent.DataTickingCard
import com.extendedae_plus.util.UtilTextComponent
import net.minecraft.core.registries.Registries
import net.minecraft.world.item.CreativeModeTab
import net.neoforged.neoforge.registries.DeferredHolder
import net.neoforged.neoforge.registries.DeferredRegister

object ModCreativeTabs {
    @InitObject
    val Register: DeferredRegister<CreativeModeTab> =
        DeferredRegister.create(Registries.CREATIVE_MODE_TAB, ExtendedAEPlus.MODID)

    val Main: DeferredHolder<CreativeModeTab, CreativeModeTab> = Register.register("main") { ->
        CreativeModeTab.builder()
            .icon(ModItems.WirelessTransceiver::toStack)
            .displayItems { _, output ->
                ModItems.Items.forEach(output::accept)

                listOf(
                    DataTickingCard(2, 16),
                    DataTickingCard(4, 192),
                    DataTickingCard(8, 512),
                    DataTickingCard(16, 1024)
                ).map(DataTickingCard::toStack).forEach(output::accept)
            }.withTabFactory {
                object : CreativeModeTab(it) {
                    override fun getDisplayName() = UtilTextComponent.modNameColorful
                }
            }.build()
    }
}
